sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeCreator.scala

```diff
@@ -82,7 +82,13 @@ case class GpuCreateNamedStruct(children: Seq[Expression]) extends GpuExpression
   }.toList.unzip
 
   private lazy val names = nameExprs.map {
-    case g: GpuExpression => g.columnarEval(null)
+    case g: GpuExpression =>
```

Comment:
It would be better
```Scala
    case gLit: GpuLiteral => gLit.value
    case ge: GpuExpression => throw new IllegalStateException("Unexpected GPU expression")
```

Reply:
Hi @firestarman, One question is how do we ensure it must be GpuLiteral? Is there any possibility that the nameExprs is some other GpuExpression?

Reply:
Yes, you are right, we need to do verification and take care of all possible cases. Calling `columnarEval` with an invalid batch (null) is dangerous .

Reply:
Yeah, But I will take your suggestion

Reply:
Great fix! To me, `ret` here can not belong to types other than `GpuScalar`.  So, we can assert it is a `GpuScalar`.

Reply:
This `columnarEval(null)` existed before, so it is not the issue of this PR, but we can update it in the PR.

Reply:
Done